### PR TITLE
Route at start of draft

### PR DESCRIPTION
**Pull request by edwardhorsford**
This moves route selection back to the start of the draft record journey.

As a byproduct, when selecting a publish course, we can now filter by route, reducing the number of choices.

## Asking about route
Screenshot 2021-02-01 at 12 35 53

## Confirming route on task list
Screenshot 2021-02-01 at 12 36 00

### CODE REVIEW

**Comment by dankmitchell** [comment]
[Prototype: ask for route at start of journey](https://trello.com/c/CpGrKam2/942-prototype-ask-for-route-at-start-of-journey)

File: app/data/training-route-data.js [hunk 1]
```diff
@@ -81,29 +81,31 @@ Object.keys(trainingRoutes).sort().forEach(function(key) {
 trainingRoutes = orderedTrainingRoutes
 
 allocatedSubjects = [
-  "Physical education"
+  "Physical education",
+  "Primary with physical education",
+  "Sport and Physical Education"
```
Comment (jesseyuen) [inline app/data/training-route-data.js]: "Primary with physical education",
"Sport and Physical Education"

These are EBACC courses right? I believe they don't fall under allocations.

https://ukgovernmentdfe.slack.com/archives/C01KM7E3E0Z/p1612261607107800

File: app/views/new-record/overview.html [hunk 2]
```diff
@@ -19,27 +21,37 @@
...
 
+{# Text shown if user belongs to multiple providers #}
+{% set providerText %}
   {# Include a link if using blended model #}
   {% if data.signedInProviders | length > 1 %}
-    {% set insetProviderText %}
-      Training with <a href="./pick-provider" class="govuk-link">{{record.provider}}</a>.
-    {% endset %}
-
+    Training with <a href="{{'./pick-provider' | addReferrer(referrer) }}" class="govuk-link">{{record.provider}}</a>
```
Comment (jesseyuen) [inline app/views/new-record/overview.html:29]: Will there be a unique URL for each draft in production? If not, we should consider adding the `govuk-link--no-visited-state` class on this link.

File: app/views/new-record/overview.html [hunk 3]
```diff
@@ -19,27 +21,37 @@
...
   {% else %}
-    {% set insetProviderText %}
-      Training with {{record.provider}}.
-    {% endset %}
+    Training with {{record.provider}}
   {% endif %}
+{% endset %}
+
+{% set routeText %}
+  on the <a href="{{ './select-route' | addReferrer(referrer) }}" class="govuk-link">{{ record.route | lower or "route not set" }}</a> route.
```
Comment (jesseyuen) [inline app/views/new-record/overview.html:37]: Same as above. If there won't be a unique url in production we should consider adding the `govuk-link--no-visited-state` class on this link.

**Review by jesseyuen — APPROVED** [review]

File: app/views/new-record/overview.html (hunk 3 above)
Comment (edwardhorsford) [reply app/views/new-record/overview.html:37]: Yes there should be.

File: app/views/new-record/overview.html (hunk 2 above)
Comment (edwardhorsford) [reply app/views/new-record/overview.html:29]: Yes there should be.

File: app/data/training-route-data.js (hunk 1 above)
Comment (edwardhorsford) [reply app/data/training-route-data.js]: Ah my bad. I can revert.

**Pull request merged**